### ISSUE TITLE
Incompatible with Java 22

### ISSUE THREAD

**Issue by RoblKyogre** [MopsTMC/C4Music#5]
Using Linux (more specifically, Arch Linux with the zen kernel at `6.9.9-zen1-1-zen`) with OpenJDK 22.0.1, the resource pack does not show in the pack selection menu, preventing me from using this resource pack. I've only tested this with 1.21 thus far, but I'll see if it is broken in 1.20.6 or not when I have the opportunity.

My guess is it has to do with the extraction protection methods used in the zip file. Opening in Ark or attempting to extract in Dolphin gives me a "Multi-disk zip archives not supported" error, and attempting to run `unzip` on the file notes that the zipfile "claims to be last disk of a multi-part archive."

**Comment by RoblKyogre** [comment]
Tested on 1.20.6, also can’t see the pack there

**Comment by MopsTMC** [comment]
[Java 22 seems like the incompatibility](https://github.com/ComunidadAylas/PackSquash/issues/294) here, can you attatch a snippet of the log to confirm?
Unfortunately if it is, I can't be of much help

**Comment by RoblKyogre** [comment]
unfortunately, i don’t think i have much of a stable internet connection for the next few days, so don’t think i’ll be able to download java 21 until it improves
in the mean time, here’s a log of it on java 22: https://mclo.gs/OVnldco

**Comment by MopsTMC** [comment]
Yeah, the errors match. I started using the zip extraction protection because the ogg obfuscation stopped working with Mojang's changes in 1.20.5, so to summarise
- < v3.2 : Incompatible with 1.20.5+
- ≥ v3.1 : Incompatible with Java 22

**Comment by Oregano1963** [comment]
Can confirm with Java 23 and 24 (EA).

**Comment by VerMishelb** [comment]
It's not Java 22. I have 21.0.5 installed and the resource pack is not being displayed in the in-game list.

**Comment by MopsTMC** [comment]
You may download the v4.1 file, compatible with Java 22+ from the v4 version page, it is listed as a secondary upload.
https://modrinth.com/resourcepack/c4music/version/4.0. I apologise for the huge delay

**Comment by AViewFromTheTop** [comment]
> You may download the v4.1 file, compatible with Java 22+ from the v4 version page, [it is listed as a secondary upload](https://modrinth.com/resourcepack/c4music/version/4.0). I apologise for the huge delay [Direct Download Link](https://cdn.modrinth.com/data/QSKCFxWX/versions/gW7HSEgE/C4Music_v4.1.zip)

Hey!
Real quick, I'd like to ask- how exactly do I do this?
I want two C418 songs to be available in my mod (1.21-1.21.6,) but I have no idea how to do it properly on top of supporting Java 22+.
- Thank you!

**Comment by MopsTMC** [comment]
Hello, the method I use to protect the pack is zip-wide since the older method of per-ogg-file obfuscation does not work on 1.20+. Minecraft does not have many problems parsing these protected packs but I would guess mod loaders will complain if not outright reject them as jars. _I do not know of any way to protect individual files._

If you mean resource pack, not mod, the tool is Packsquash ([specifically this option](https://github.com/ComunidadAylas/PackSquash/wiki/Options-files#zip_spec_conformance_level)), but as mentioned, this is going to obfuscate the zip file

**Comment by AViewFromTheTop** [comment]
> Hello, the method I use to protect the pack is zip-wide since the older method of per-ogg-file obfuscation does not work on 1.20+. Minecraft does not have many problems parsing these protected packs but I would guess mod loaders will complain if not outright reject them as jars. _I do not know of any current way to protect individual files._
> 
> If you mean resource pack, not mod, the tool is Packsquash ([specifically this option](https://github.com/ComunidadAylas/PackSquash/wiki/Options-files#zip_spec_conformance_level)), but as mentioned, this is going to obfuscate the entire zip file

Ye! I meant the zip!
Sadly last night I found out that gradle (used for mod projects) doesn’t like using obfuscated resource pack jars… it always gives an error. I’m not really sure what to do, but maybe one day!
I appreciate it!

**Comment by MopsTMC** [comment]
Well, in the project's author's own words this method breaks the zip file just enough to not compatible with anything other than Minecraft. Good luck with your endeavor though!

**Comment by AViewFromTheTop** [comment]
Took it upon myself to make an API for this today, now I can have obfuscated jars in my mods and it works great!
Maybe a bit cheat-y but it seems to be flawless.
Thank you!!

**Comment by MopsTMC** [comment]
Interesting approach, I'm glad it works
